lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -90,6 +95,8 @@ class DeadFunctionAndGlobalElimination {
 
   bool keepExternalWitnessTablesAlive;
 
+  bool removeUnspecializedFunctionsInEmbeddedSwift;
```

Comment:
I'm not sure if we actually need this flag.
The logic is that until the swiftmodule file is written, i.e. the SerializeSILPass runs, we still need to keep generic functions. But then we can delete them.
For checking if the module was already serialized there is `SILModule::isSerialized()`